### ISSUE TITLE
Is it possible to integrate a pytorch module into hasktorch?

### ISSUE THREAD

**Issue by Abastro** [hasktorch/hasktorch#542]
Hello, looking into using this library for my small project.
I want to know, if it is possible to integrate a pytorch module, simply calculating some tensor & gradients, into hasktorch?
I think I can make the throughput of the data small enough to be easily passed around, which is of course not ideal. I wonder if there would be more streamlined way.
Main reasoning is that it would be painful to translate the entire python code into haskell myself.
Also please let me know where to ask this kind of questions, in case issue is not an appropriate place

**Comment by junjihashimoto** [comment]
Hi,
If the pytorch module support torchscript, it may be possible.
https://pytorch.org/docs/stable/jit.html

https://github.com/hasktorch/hasktorch/blob/master/hasktorch/test/ScriptSpec.hs#L73-L95
This example calculates gradients of torchscript in hasktorch.

**Comment by Abastro** [comment]
Thank you! I want to ask, how do I check if a module could be scripted with torchscript? Maybe I would have to , because the code is from some paper.

**Comment by junjihashimoto** [comment]
To check it, just call `torch.jit.script("your module"())` .
When the product does not support torchscript, it will give an error.
torchscript is a static type, and cases where the output type is different between inference and training will not work.

**Comment by Abastro** [comment]
What I wanted to use was "Sparse Convolution", which uses a new type "SparseConvTensor". Is there no way to use it then? I thought there would be a way to use FFI toward the python code.

**Comment by junjihashimoto** [comment]
Is the "Sparse Convolution" https://github.com/traveller59/spconv?
It's not impossible to use it, but it is difficult, because the codes have python and c++.
You need to register all the c ++ functions that torchscript does not have.
torchvision's codes will be helpful.

**Comment by junjihashimoto** [comment]
The registration for ops is here.
https://pytorch.org/tutorials/advanced/torch_script_custom_ops.html

**Comment by Abastro** [comment]
How do I register the operations in hasktorch? Also how can I make it run with my own CUDA code?
